frontend/packages/vkt/src/utils/enrollment.ts

```diff
@@ -25,4 +30,15 @@ export class EnrollmentUtils {
       isUnderstandingExamsSelected
     );
   }
+
+  static isValidCertificateShipping(shippingData: CertificateShippingData) {
+    const hasNonBlankFieldValues = [
```

Comment:
Booleanit voisi pitää lähtökohtaisesto ilman negaatiota. hasNonBlankFieldValues => `isEveryFieldFilled` tai `isEachFieldFilled`. Tällee ei vahingossa myöhemmin joudu pyöritteleen tuplanegaatioita `!hasNonBlankFieldValues`

Samalla voisi tsekata noi yleisesti käytetyt funkkarit niin ehkä omanssa PR:ssä. Natiivisesti esim. `isEmpty()` implementaatio lodash / underscore kirjastoista: https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_isempty

Noi voinee sit korvata jos toi lodash napataan fronttiin käyttöön

Reply:
Joo, ja aika paljon kaikkea muutakin joita voisi tuossa lodash-PR:ssä toteuttaa tai sitten myöhemmin kun ko. koodiin koskee. Toi lodashin käyttöönottokeskustelu lähti oikeastaan siitä, kun havaittiin että joudutaan paikoin tekemään turhaa filteröintiä siten, että jaetaan jokin taulukko esim. tietyn kentän boolean-arvon perusteella. Ts. ei pysty kirjoittamaan mitään tällaista https://www.tutorialspoint.com/scala_collections/scala_collections_partition.htm vaan sen sijaan 2 eri filteröintiä samalle listalle.